gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagerUtils.java

```diff
@@ -48,25 +53,46 @@
 import org.apache.gobblin.service.modules.flowgraph.Dag.DagNode;
 import org.apache.gobblin.service.modules.orchestration.DagManager.FailureOption;
 import org.apache.gobblin.service.modules.spec.JobExecutionPlan;
+import org.apache.gobblin.service.monitoring.JobStatus;
+import org.apache.gobblin.service.monitoring.JobStatusRetriever;
 import org.apache.gobblin.util.ConfigUtils;
 
+import static org.apache.gobblin.service.ExecutionStatus.PENDING;
+
 
 public class DagManagerUtils {
   static long DEFAULT_FLOW_SLA_MILLIS = TimeUnit.HOURS.toMillis(24);
   static String QUOTA_KEY_SEPERATOR = ",";
 
-  static FlowId getFlowId(Dag<JobExecutionPlan> dag) {
+  public static FlowId getFlowId(Dag<JobExecutionPlan> dag) {
     return getFlowId(dag.getStartNodes().get(0));
   }
 
+  public static DagActionStore.DagAction createDagAction(Dag<JobExecutionPlan> dag, DagActionStore.FlowActionType flowActionType) {
+    return createDagAction(dag.getStartNodes().get(0), flowActionType);
+  }
+
+  public static DagActionStore.DagAction createDagAction(String flowGroup, String flowName, String flowExecutionId, DagActionStore.FlowActionType flowActionType) {
+    return new DagActionStore.DagAction(flowGroup, flowName, flowExecutionId, flowActionType);
+  }
```

Comment:
I don't see this adding value (i.e. `createDagAction` vs `new DagAction`). do we really need it?